javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java

```diff
@@ -154,11 +154,21 @@ protected void writeMethodImplementation(
         // invocation with protocol method parameters and RequestOptions
         String invocationExpression = protocolMethod.getMethodInputParameters().stream()
                 .map(p -> {
-                    String expression = parameterExpressionsMap.get(p.getName());
-                    if (expression == null) {
-                        expression = p.getName();
+                    String parameterName = p.getName();
+                    String expression = parameterExpressionsMap.get(parameterName);
+                    IType parameterClientType = p.getClientType();
+                    IType parameterRawType = p.getRawType();
+
+                    if (ClientModelUtil.isClientModel(parameterRawType) && ClientModelUtil.isJsonMergePatchModel(ClientModelUtil.getClientModel(((ClassType) parameterRawType).getName()))) {
```

Comment:
nit, also add a check for `RequestParameterLocation.BODY`?

If possible, add a test case when this request body is optional.

Reply:
I will add the request body optional case into cadl-ranch.

Reply:
Seems not much code change, I updated together in this pr.

Reply:
OK, then it be great.

I wasn't expect these old code could handle that without any modification outside of this block, but appears it does.